src/test/java/org/opensearch/security/auth/UserInjectorTest.java

```diff
@@ -64,6 +64,44 @@ public void testValidInjectUser() {
         assertEquals(injectedUser.getRoles(), roles);
     }
 
+    @Test
+    public void testValidInjectUserIpV6() {
+        HashSet<String> roles = new HashSet<>();
+        roles.addAll(Arrays.asList("role1", "role2"));
+        threadContext.putTransient(
+            ConfigConstants.OPENDISTRO_SECURITY_INJECTED_USER,
+            "user|role1,role2|2001:db8:3333:4444:5555:6666:7777:8888:9200"
+        );
+        User injectedUser = userInjector.getInjectedUser();
+        assertEquals(injectedUser.getName(), "user");
```

Comment:
Can we also add assertions to ensure that it parses the expected port?

Reply:
Done in: [7120a8d](https://github.com/opensearch-project/security/pull/4399/commits/7120a8d21d8805ec9d9240ce8d65a1436d45b85a)